biorxiv/time_to_publication/3.0-mjt-kaplan-meier-plots.py

```diff
@@ -0,0 +1,236 @@
+merged_df.loc[~merged_df["published"], "observation_date"] = pd.datetime.today()
+
+
+# In[18]:
+
+
+merged_df["observation_duration"] = (
+    merged_df["observation_date"] - merged_df["date_received"]
+)
+
+
+# In[19]:
+
+
+(merged_df["observation_duration"] < pd.Timedelta(0)).sum()
+
+
+# In[20]:
+
+
+merged_df = merged_df[merged_df["observation_duration"] > pd.Timedelta(0)]
+
+
+# In[21]:
+
+
+ax = sns.distplot(merged_df["observation_duration"].dt.total_seconds() / 60 / 60 / 24 / 365)
+
+
+# In[22]:
+
+
+kmf = KaplanMeierFitter()
+
+
+# In[23]:
+
+
+kmf.fit(
+    merged_df["observation_duration"].dt.total_seconds() / 60 / 60 / 24 / 365,
+    event_observed=merged_df["published"],
+)
+ax = kmf.plot(label="all papers", logx=True)
+_ = ax.set_ylabel("proportion of unpublished biorxiv papers")
+_ = ax.set_xlabel("timeline (years)")
+_ = ax.set_ylim(0, 1)
+
+
+# In[24]:
+
+
+f = plt.figure(figsize=(10, 8))
+
+ax = None
+for category, cat_group in merged_df.groupby("category"):
+    kmf.fit(
+        cat_group["observation_duration"].dt.total_seconds() / 60 / 60 / 24 / 365,
+        event_observed=cat_group["published"],
+    )
+    ax = kmf.plot(label=category, ax=ax, ci_show=False, logx=True)
+
+# Shrink current axis by 20%
+box = ax.get_position()
+ax.set_position([box.x0, box.y0, box.width * 0.8, box.height])
+
+# Put a legend to the right of the current axis
+_ = ax.legend(loc="center left", bbox_to_anchor=(1, 0.5), title="Biorxiv category")
+
+_ = ax.set_ylabel("proportion of unpublished biorxiv papers")
+_ = ax.set_xlabel("timeline (years)")
+_ = ax.set_ylim(0, 1)
+
+
+# In[25]:
+
+
+merged_df["doi_prefix"] = merged_df["published_doi"].str.split("/").str[0]
+
+
+# In[26]:
+
+
+get_ipython().run_cell_magic('time', '', 'f = plt.figure(figsize=(10, 8))\n\nax = None\nfor category, cat_group in merged_df.groupby("doi_prefix"):\n    if len(cat_group) > 100:\n        kmf.fit(\n            cat_group["observation_duration"].dt.total_seconds() / 60 / 60 / 24 / 365,\n            event_observed=cat_group["published"],\n        )\n        ax = kmf.plot(label=category, ax=ax, ci_show=False, logx=True)\n\n# Shrink current axis by 20%\nbox = ax.get_position()\nax.set_position([box.x0, box.y0, box.width * 0.8, box.height])\n\n# Put a legend to the right of the current axis\n_ = ax.legend(loc="center left", bbox_to_anchor=(1, 0.5), title="DOI prefix")\n\n_ = ax.set_ylabel("proportion of unpublished biorxiv papers")\n_ = ax.set_xlabel("timeline (years)")\n_ = ax.set_ylim(0, 1)')
+
+
+# In[27]:
+
+
+get_ipython().run_cell_magic('time', '', 'doi_prefix_df = merged_df.groupby("doi_prefix").apply(\n    lambda cat_group: pd.Series(\n        {\n            "count": len(cat_group),\n            "80th_percentile": kmf.fit(\n                cat_group["observation_duration"].dt.total_seconds()\n                / 60\n                / 60\n                / 24,\n                event_observed=cat_group["published"],\n            ).percentile(0.8),\n        }\n    )\n)')
```

Comment:
Can you clarify the 80th percentile? Is this the number of papers published grouped by publication time?

Reply:
Sorry, 80th percentile is a horizontal line at .8 in the plot above. I was trying to identify the pink and green lines in the plot above. 80th percentile roughly translates to the number of days the first 20% of papers published in the journal are published in. f1000 is fastest at 27 days, i.e. 20 percent of preprints that eventually end up in f1000 are published in less than a month of posting on biorxiv. One of the big strategy initiatives for Elsevier is improving time between submission and publication.